### ISSUE TITLE
Docker Provider Freezing On Init

### ISSUE THREAD

**Issue by LTheobald** [loft-sh/devpod#417]
**What happened?**
When creating a Docker provider on my Intel based Mac, the provider would always freeze. The GUI would sit there spinning it's wheel, the CLI would never complete.

**What did you expect to happen instead?**
A docker provider to be created

**How can we reproduce the bug?** (as minimally and precisely as possible)
I was going from a fresh DevPod install on a Intel Mac; zsh as my shell; the docker command is an alias to `multipass docker` which I use to run Docker on my Mac.

My `devcontainer.json`:
None. No workspaces created yet.

**Local Environment:**

* DevPod Version: v0.1.9
* Operating System: Mac
* ARCH of the OS: x86

**DevPod Provider:**
Local Docker provider.
Standard didn't work; [gist with changed init script did](https://gist.githubusercontent.com/LTheobald/f91715c9eb37db3da3416ad841b384f0/raw/ab17aae13b8f0f8e8487eca6e5f1f733fa813801/provider.yaml)

**Anything else we need to know?**
Upon debugging, the issue for me was the command the init script used to test if Docker was running. This was never completing:

```
DOCKER_REACHABLE=$(${DOCKER_PATH} ps > /dev/null 2>&1 && echo -n "true" || echo -n "false")
```

Switching the init script instead to use (excuse my poor shell scripting):

```
${DOCKER_PATH} ps
if [ $? -eq 0 ]; then
  echo "Docker is running"
else
  >&2 echo "Seems like docker is not reachable on your system."
  >&2 echo "Please make sure docker is installed and running."
  >&2 echo "You can verify if docker is running correctly via '${DOCKER_PATH} ps'"
  exit 1
fi
```

And the init script complete fine.

Interestingly, although I used a gist with the provider set at v0.0.2, it's still showing as v0.0.1 in list.

Provider: [https://gist.githubusercontent.com/LTheobald/f91715c9eb37db3da3416ad841b384f0/raw/ab17aae13b8f0f8e8487eca6e5f1f733fa813801/provider.yaml](https://gist.githubusercontent.com/LTheobald/f91715c9eb37db3da3416ad841b384f0/raw/ab17aae13b8f0f8e8487eca6e5f1f733fa813801/provider.yaml)

```
devpod provider list
  
   NAME  | VERSION | DEFAULT | INITIALIZED |   DESCRIPTION     
---------+---------+---------+-------------+-------------------
  docker | v0.0.1  | true    | true        | DevPod on Docker  
```

Will attempt to dig into this more later to see if it's just me or not.

**Comment by benborges** [comment]
getting the exact same issue on my arch system :/

**Comment by benborges** [comment]
Well...I ran docker-desktop as a user and then added the docker provider and it went smooth.

**Comment by FabianKramm** [comment]
@benborges @LTheobald we used your fix and added it via #421, if that doesn't work, feel free to reopen please
